src/main/cpp/locationinfo.cpp

```diff
@@ -45,16 +45,35 @@ LocationInfo::LocationInfo( const char* const fileName1,
 	int lineNumber1 )
 	:  lineNumber( lineNumber1 ),
 	   fileName( fileName1 ),
+	   shortFileName(nullptr),
 	   methodName( methodName1 )
 {
 }
 
+LocationInfo::LocationInfo( const char* const fileName1,
+							size_t shortFilenameOffset,
+	const char* const methodName1,
+	int lineNumber1 )
+	:  lineNumber( lineNumber1 ),
+	   fileName( fileName1 ),
+	   methodName( methodName1 )
+{
+	if( shortFilenameOffset < 0 || shortFilenameOffset > 300 ){
```

Comment:
My (perhaps misguided) reasoning is that because this accesses a `const char*` directly, that could let you access arbitrary memory.  I'm not sure how likely that is though.

Reply:
If the file name calculation works, there won't be such access. If it doesn't work, it could easily happen at 199 chars as well already. OTOH, arbitrary path restrictions regarding their length, allowed characters like spaces etc. are a continuing annoyance for decades, so in my opinion we shouldn't add one unless absolutely necessary.